src/utils/variableType.js

```diff
@@ -47,7 +66,7 @@ function isDeclaredVariable(node, parent) {
 }
 
 function isPropertyInMemberExpression(node, parent) {
-  return parent.type === 'MemberExpression' && parent.property === node;
+  return parent.type === 'MemberExpression' && parent.property === node && !parent.computed;
```

Comment:
Wow... you fixed a bug in `let` transform.

I should also add a unit test for this, which goes something like so:

```js
expectNoChange(
  'if (true) {\n' +
  '  var x = 0;\n' +
  '}\n' +
  'foo[x]++;'
);
```